cpp/benchmarks/io/text/multibyte_split.cpp

```diff
@@ -102,30 +102,30 @@ static void bench_multibyte_split(nvbench::state& state)
 
   auto const delim_factor = static_cast<double>(delim_percent) / 100;
   auto device_input       = create_random_input(file_size_approx, delim_factor, 0.05, delim);
-  auto host_input         = thrust::host_vector<char>(device_input.size());
-  auto const host_string  = std::string(host_input.data(), host_input.size());
-
-  cudaMemcpyAsync(host_input.data(),
-                  device_input.data(),
-                  device_input.size() * sizeof(char),
-                  cudaMemcpyDeviceToHost,
-                  cudf::default_stream_value);
-
-  auto const temp_file_name = random_file_in_dir(temp_dir.path());
-
-  {
-    auto temp_fostream = std::ofstream(temp_file_name, std::ofstream::out);
-    temp_fostream.write(host_input.data(), host_input.size());
-  }
-
-  cudaDeviceSynchronize();
+  auto host_input         = thrust::host_vector<char>();
 
   auto source = [&] {
+    // copy data to host if necessary
+    if (source_type != data_chunk_source_type::device) {
+      host_input.resize(device_input.size());
+      cudaMemcpyAsync(host_input.data(),
+                      device_input.data(),
+                      device_input.size() * sizeof(char),
+                      cudaMemcpyDeviceToHost,
+                      cudf::default_stream_value);
+      cudf::default_stream_value.synchronize();
+    }
```

Comment:
Let's move this in to a helper function and call it more locally so we don't duplicate the conditional logic.
```suggestion
    // copy data to host if necessary
    auto get_host_input = [&](){
      auto host_input = thrust::host_vector<char>(device_input.size());
      CUDF_CUDA_TRY(cudaMemcpyAsync(host_input.data(),
                      device_input.data(),
                      device_input.size() * sizeof(char),
                      cudaMemcpyDeviceToHost,
                      cudf::default_stream_value));
      cudf::default_stream_value.synchronize();
      return host_input;
    };
```

Reply:
[I think we should try to leverage the vector-factories here instead.
https://github.com/rapidsai/cudf/blob/8b9f203e0fa6e41f1d27724feba424f5caeb0968/cpp/include/cudf/detail/utilities/vector_factories.hpp#L201-L210](https://github.com/rapidsai/cudf/blob/8b9f203e0fa6e41f1d27724feba424f5caeb0968/cpp/include/cudf/detail/utilities/vector_factories.hpp#L345-L351)